cpp/src/reductions/scan.cu

```diff
@@ -8,9 +8,11 @@
 
 #include <cub/device/device_scan.cuh>
 
+#include "reduction_operators.cuh"
+
 namespace { //anonymous
 
-#define COPYMASK_BLOCK_SIZE 1024
+static constexpr int copy_replace_nulls_blocksize = 1024;
```

Comment:
Seems pretty big... consider using CUDA occupancy API...

Reply:
Yes, update to use CUDA occupancy API